### ISSUE TITLE
Better variation stock management

### ISSUE THREAD

**Issue by f3bruary** [woocommerce/woocommerce#14493]
### EXPLANATION OF THE FEATURE REQUEST

Example case: e-Commerce selling shoes.

I buy an assortment of a certain shoe model. It comes in 12 pairs. Sizes (EU): 36x1, 37x2, 38x3, 39x3, 40x2, 41x1.

Stock is managed on variation level. As a seller, it is not very relevant to me how many of each size are left in stock, low on stock or out of stock. It is, however, relevant to know how many pairs of a certain product I have left. Currently this is not possible.

The parent (which is irrelevant when the product type is variable, aside from main attributes) should have the total amount of variations as stock level. So the parent product here should have a stock of 12. This way the low stock threshold is much more practical to keep an eye on. I'd know if I sold a certain model shoe fast and if it's early in the season this could convince me to buy more. If a certain model hasn't sold any then it's time to add a discount.

Basically the whole system would be much more convenient. My dashboard widget now shows 2000 products that are low on stock, and 2000 that are out of stock. These numbers are useless to me as I have no idea if a certain model is actually sold out or low on stock. I'm also forced to disable the stock alerts as it would only spam me for each variations that's sold out or low on stock. Something which I would very much like to enable.

**Comment by mikejolley** [comment]
As product schemas are optimised this will be more possible, right now however this would be very query intensive. 

I've cross-linked this with https://github.com/woocommerce/woocommerce/issues/13086 so when the variations are redone this is considered.
